packages/contracts/test/transactionManager.spec.ts

```diff
@@ -299,20 +315,36 @@ describe("TransactionManager", function () {
     // Generate signature from user
     const signature = await signFulfillTransactionPayload(transaction, relayerFee, user);
 
+    const invariantDigest = await getInvariantTransactionDigest(transaction);
+    const variantDigestBeforeFulfill = await getVariantTransactionDigest({
+      amount: record.amount,
+      expiry: record.expiry,
+      preparedBlockNumber: record.preparedBlockNumber,
+    });
+
+    expect(await instance.variantTransactionData(invariantDigest)).to.be.eq(variantDigestBeforeFulfill);
     // Send tx
-    const tx = await instance
-      .connect(submitter)
-      .fulfill(
-        { ...transaction, amount: record.amount, expiry: record.expiry, blockNumber: record.blockNumber },
-        relayerFee,
-        signature,
-      );
+    const tx = await instance.connect(submitter).fulfill(
+      {
+        ...transaction,
+        ...record,
+      },
+      relayerFee,
+      signature,
+    );
     const receipt = await tx.wait();
     expect(receipt.status).to.be.eq(1);
 
+    const variantDigest = await getVariantTransactionDigest({
+      amount: record.amount,
+      expiry: record.expiry,
+      preparedBlockNumber: 0,
+    });
+
+    expect(await instance.variantTransactionData(invariantDigest)).to.be.eq(variantDigest);
     // Assert event
     await assertReceiptEvent(receipt, "TransactionFulfilled", {
-      txData: { ...transaction, amount: record.amount, expiry: record.expiry, blockNumber: record.blockNumber },
+      txData: { ...transaction, ...record },
```

Comment:
We should make sure we have tests with actual relayers, not sure if we do

Reply:
we don't have one, but I don't think if that is necessary for the contract unit test